### PR TITLE
make dependency on primme optional

### PR DESCRIPTION
**Pull request by liwt31**
New users of renormalizer often struggle to install primme, which is not frequently used. In this PR I make the dependency on `primme` optional. It is removed from `requiements.txt`. But if it is correctly installed, renormalizer still supports the package. The Travis CI test configurations are updated such that two different sets of tests are run. The first is without `primme` (default) and the second is with `primme`. 

Also fix a small bug in OFS.

### CODE REVIEW

**Comment by liwt31** [comment]
It seems we have run out of credit on Travis CI. So I switch to Circle CI.

File: requirements.txt [hunk 1]
```diff
@@ -7,5 +7,4 @@ PyYAML==5.4.*
 recommonmark==0.7.*
 sphinx_rtd_theme==0.5.*
 opt_einsum==3.3.*
-primme==3.2.*
```
Comment (jjren) [inline requirements.txt:10]: It is better to keep the `primme` package here but with a comment  # in order to remind the user to install it if possible.

Comment (liwt31) [reply requirements.txt:10]: I don't think a comment here will help "remind" any user. It is better added to the installation doc, and the old one is very outdated.

Comment (jjren) [reply requirements.txt:10]: at least remind us :joy:

**Review by jjren — APPROVED** [review]

**Pull request merged**